src/gui/qgsmaskingwidget.cpp

```diff
@@ -107,15 +87,15 @@ QList<QPair<QString, QList<QgsSymbolLayerReference>>> symbolLayerMasks( const Qg
 
 void QgsMaskingWidget::setLayer( QgsVectorLayer *layer )
 {
-  if ( mLayer != layer )
-  {
-    mLayer = layer;
-    mMustPopulate = true;
-  }
+  mLayer = layer;
+  populate();
 }
 
 void QgsMaskingWidget::populate()
 {
+  mMaskSourcesWidget->blockSignals( true );
+  mMaskTargetsWidget->blockSignals( true );
+
```

Comment:
How about using a scoped signal blocker here, for safety and less verbosity ?
See https://doc.qt.io/qt-6/qsignalblocker.html

Reply:
Good idea, done